src/Resources/config/core.php

```diff
@@ -48,7 +48,7 @@
         ->set('sonata.admin.pool', Pool::class)
             ->public()
             ->args([
-                new ReferenceConfigurator('service_container'),
+                null, // admin service locator
```

Comment:
Is it needed to keep the `args` call ?
I find weird to pass `null` when it's not supported by the constructor

Reply:
> Is it needed to keep the `args` call ?

It is in order to be able to use `$pool->replaceArgument(0, ServiceLocatorTagPass::register($container, $adminServices));`

> I find weird to pass `null` when it's not supported by the constructor

It is a common pattern when using Compiler Passes, see https://github.com/symfony/symfony/blob/3fca96d9de706f9da12b488e05d395444a71d9c7/src/Symfony/Bundle/FrameworkBundle/Resources/config/form.xml#L37 for example. In Symfony 5.1 there is an [`AbstractArgument`](https://github.com/symfony/symfony/blob/3fca96d9de706f9da12b488e05d395444a71d9c7/src/Symfony/Component/DependencyInjection/Argument/AbstractArgument.php) for this, but we cannot use it yet.